public/components/custom_panels/helpers/utils.tsx

```diff
@@ -210,130 +211,11 @@ export const getQueryResponse = (
-  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
-  setIsError: React.Dispatch<React.SetStateAction<VizContainerError>>
-) => {
-  setIsLoading(true);
-  setIsError({} as VizContainerError);
-
-  let visualization: SavedVisualizationType = {};
-  let updatedVisualizationQuery = '';
-
-  visualization = await fetchVisualizationById(http, savedVisualizationId, setIsError);
-
-  if (_.isEmpty(visualization)) {
-    setIsLoading(false);
-    return;
-  }
-
-  if (visualization.name) {
-    setVisualizationTitle(visualization.name);
-  }
-
-  if (visualization.type) {
-    setVisualizationType(visualization.type);
-  }
-
-  if (spanParam !== undefined) {
-    updatedVisualizationQuery = updateQuerySpanInterval(
-      visualization.query,
-      visualization.timeField,
-      spanParam
-    );
-  } else {
-    updatedVisualizationQuery = visualization.query;
-  }
-
-  setVisualizationMetaData({ ...visualization, query: updatedVisualizationQuery });
-
-  getQueryResponse(
-    pplService,
-    updatedVisualizationQuery,
-    visualization.type,
-    startTime,
-    endTime,
-    setVisualizationData,
-    setIsLoading,
-    setIsError,
-    filterQuery,
-    visualization.timeField
-  );
-};
-
-const createCatalogVisualizationMetaData = (
-  catalogSource: string,
-  visualizationQuery: string,
-  visualizationType: string,
-  visualizationTimeField: string
-) => {
-  return {
-    name: catalogSource,
-    description: '',
-    query: visualizationQuery,
-    type: visualizationType,
-    selected_date_range: {
-      start: 'now/y',
-      end: 'now',
-      text: '',
-    },
-    selected_timestamp: {
-      name: visualizationTimeField,
-      type: 'timestamp',
-    },
-    selected_fields: {
-      text: '',
-      tokens: [],
-    },
-  };
-};
-
-const updateCatalogVisualizationQuery = ({
-  catalogSourceName,
-  catalogTableName,
-  aggregation,
-  attributesGroupBy,
-  startTime,
-  endTime,
-  spanParam,
-}: {
-  catalogSourceName: string;
-  catalogTableName: string;
-  aggregation: string;
-  attributesGroupBy: string[];
-  startTime: string;
-  endTime: string;
-  spanParam: string | undefined;
-}) => {
-  const attributesGroupString = attributesGroupBy.toString();
-  const startEpochTime = convertDateTime(startTime, true, false, true);
-  const endEpochTime = convertDateTime(endTime, false, false, true);
-  // const promQuery =
-  //   attributesGroupBy.length === 0
-  //     ? `${aggregation} (${catalogTableName})`
-  //     : `${aggregation} by(${attributesGroupString}) (${catalogTableName})`;
-
-  const promQuery = `${aggregation} (${catalogTableName})`;
-
-  return `source = ${catalogSourceName}.query_range('${promQuery}', ${startEpochTime}, ${endEpochTime}, '${spanParam}')`;
-};
-
-// Creates a catalogVisualization for a runtime catalog based PPL query and runs getQueryResponse
-export const renderCatalogVisualization = async ({
+export const renderSavedVisualization = async ({
```

Comment:
Git diff split this one wierdly.
`renderSavedVisualization` is retained, diff is below.
`renderCatalogVisualization`, `updateCatalogVisualizationQuery`, `createCatalogVisualizationMetaData` are removed entirely, as they are specific to PromQL query processing, which is no longer needed.

Reply:
why is it no longer needed?